provider/ec2/environ_test.go

```diff
@@ -259,30 +259,16 @@ func (*Suite) TestSelectSubnetIDsForZoneWithIncorrectPlacement(c *gc.C) {
 	c.Assert(err, gc.ErrorMatches, `subnets "boom" in AZ "foo" not found`)
 }
 
-func (*Suite) TestSelectSubnetIDsForZoneWithINFAN(c *gc.C) {
-	subnetZones := map[corenetwork.Id][]string{
-		corenetwork.Id("bar-INFAN-test"): {"foo"},
-		corenetwork.Id("baz"):            {"foo"},
-	}
-	placement := corenetwork.Id("")
-	az := "foo"
-
-	var env *environ
-	subnets, err := env.selectSubnetIDsForZone(subnetZones, placement, az)
-	c.Assert(err, jc.ErrorIsNil)
-	c.Assert(subnets, gc.DeepEquals, []corenetwork.Id{"baz"})
-}
-
 func (*Suite) TestSelectSubnetIDForInstance(c *gc.C) {
 	ctrl := gomock.NewController(c)
 	defer ctrl.Finish()
 
 	mockContext := NewMockProviderCallContext(ctrl)
 
-	subnetZones := []map[corenetwork.Id][]string{{
-		corenetwork.Id("bar-INFAN-test"): []string{"foo"},
```

Comment:
And that means we don't test the filtering in this method. It will already be filtered out in any passed arguments.